### ISSUE TITLE
Inconsistencies in Accounts* REST Endpoint 

### ISSUE THREAD

**Issue by otsakir** [RestComm/RestComm-Core#898]
Restcomm offers two set's of methods for accessing Account resources as JSON or XML. Here are the *current* URLs for all methods available:

#### AccountsJsonEndpoint.java

1 - **retrieve all** (JSON)
GET /Accounts.json

2 - **retrieve one** (JSON)
GET /Accounts.json/{accountSid}

3 - **delete**
DELETE /Accounts.json/{accountSid}.json

4 - **create** (JSON)
POST /Accounts.json

#### AccountsXmlEndpoint.java

5 - **retrieve all** (XML)
GET /Accounts

6 - **retrive one** (XML)
GET /Accounts/{accountSid}

7 - **delete** (XML)
DELETE /Accounts/{accountSid}

8 - **create** (XML)
POST /Accounts

9 - **update** (JSON)
POST /Accounts/{accountSid}.json
PUT /Accounts/{accountSid}.json

10 - **update** (XML)
POST /Accounts/{accountSid}
PUT /Accounts/{accountSid}

#### Suggestions

* The separation into AccountsXmlEndpoint and AccountsJSONEndpoint doesn't make sense and is confusing since the methods are mixed up. We should either split methods correctly or if JAX-RS doesn't permit this put them all in a single class.
* **.json** suffix should be placed in a single place, at the end. In that sence (2) and (3) are wrong. Twilio puts the suffix at the end:
https://www.twilio.com/docs/api/rest/account

Here is how the propsed endpoint would be like:

#### AccountsRestEndpoint.java:
Path: */Accounts*

1 - **retrieve all** (JSON)
GET /Accounts.json

2 - **retrieve one** (JSON)
GET /Accounts/{accountSid}.json

3 - **delete** (JSON)
DELETE /Accounts/{accountSid}.json

4 - **create** (JSON)
POST /Accounts.json

5 - **update** (JSON)
POST /Accounts/{accountSid}.json
PUT /Accounts/{accountSid}.json

6 - **retrieve all** (XML)
GET /Accounts

7 - **retrive one** (XML)
GET /Accounts/{accountSid}

8 - **delete** (XML)
DELETE /Accounts/{accountSid}

9 - **create** (XML)
POST /Accounts

10 - **update** (XML)
POST /Accounts/{accountSid}
PUT /Accounts/{accountSid}

@gvagenas thoughts ?

**Comment by ghjansen** [comment]
@otsakir i agree with this refactoring effort since I faced the difficult of understanding twice: once when i've started developing and also when i've tried to explain how it works to @FerUy in the training.

Also, some endpoints (like Clients and Gateways) have the "delete" method and its DAO logic placed inside the *XmlEndpoint.java class. There is a special need for that? I think this methods could move to its respective DAO class (like MybatisClientsDao and MybatisGatewaysDao, respectively) and leave the endpoint just to specify the entries.

**Comment by otsakir** [comment]
thanks @ghjansen

Please create additional "Suggestion"/"Tech Debt" issues if you have something in mind.

**Comment by muhammadbilal19** [comment]
While implementing the API i was also thinking the same thing why we need to use two different classes for json and xml. I recently implement this feature in one of my project, this thing can be achieve using Content Negotiation. I already implement this functionality using RestEasy API it is quite easy and straight forward in it.  
https://docs.jboss.org/resteasy/docs/3.0.6.Final/userguide/html/JAX-RS_Content_Negotiation.html

As in RestComm application we are using JAX-RS implementation using jersey and we are currently using old version of jersey, in newer version (2.0+) implementation is same as RestEasy but in older version there is some other way to implement this feature. I found a very useful link to implement this feature in jersey API https://gist.github.com/sathish06/6587432
I will give it try and will let you guys update about it

**Comment by gvagenas** [comment]
Actually the tech debt exists in other endpoints also and we should have a uniform handle of json/xml accross the whole API
